### PR TITLE
Add tests for pkg_deb output file names

### PR DESCRIPTION
**Pull request by aiuto**
- Make sure that the changes file correctly matches the file basename of the .deb file
- Add the ability to test for expected names
- Do some cleanup of deb tests in general
   - move dummy inputs from tests/testsdata to tests/deb

cc: @motiejus

### CODE REVIEW

**Comment by aiuto** [comment]
This is also ready for review again.

File: tests/deb/deb_tests.bzl [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    env = analysistest.begin(ctx)
+    target_under_test = analysistest.target_under_test(env)
+
+    ogi = target_under_test[OutputGroupInfo]
+
+    deb_path = ogi.deb.to_list()[0].path
+
+    # Test that the .changes file is computed correctly
+    changes_path = ogi.changes.to_list()[0].path
+    expected_changes_path = deb_path[0:-3] + "changes"
```
Comment (nacl) [inline tests/deb/deb_tests.bzl:28]: Maybe use `paths.replace_extension` from skylib? https://github.com/bazelbuild/bazel-skylib/blob/6e30a77347071ab22ce346b6d20cf8912919f644/docs/paths_doc.md

File: tests/deb/deb_tests.bzl [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        asserts.equals(
+            env,
+            deb_path.split("/")[-1],  # basename(path)
+            ctx.attr.expected_name,
+            "Deb package file name is not correct",
+        )
+    return analysistest.end(env)
+
+
+package_naming_test = analysistest.make(
```
Comment (nacl) [inline tests/deb/deb_tests.bzl:47]: Maybe prefix this with "deb_"?

**Review by nacl — APPROVED** [review]
Minor suggestions only; LGTM otherwise.

File: tests/deb/deb_tests.bzl (hunk 1 above)
Comment (aiuto) [reply tests/deb/deb_tests.bzl:28]: I'm OK with this as is. The extension can only be ".deb".  Anything else is badly specified.

File: tests/deb/deb_tests.bzl (hunk 2 above)
Comment (aiuto) [reply tests/deb/deb_tests.bzl:47]: I think I want the generic name. So, each type of package can have an identically named package_naming_test, differentiated by the test/<pkg type>

**Pull request merged**